bundles/org.openhab.binding.argoclima/src/main/java/org/openhab/binding/argoclima/internal/configuration/ArgoClimaConfigurationLocal.java

```diff
@@ -0,0 +1,222 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.argoclima.internal.configuration;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.argoclima.internal.ArgoClimaBindingConstants;
+import org.openhab.binding.argoclima.internal.exception.ArgoConfigurationException;
+
+/**
+ * The {@link ArgoClimaConfigurationLocal} class extends base configuration parameters with ones specific
+ * to local connection (including a remote API stub / proxy)
+ *
+ * @author Mateusz Bronk - Initial contribution
+ */
+@NonNullByDefault
+public class ArgoClimaConfigurationLocal extends ArgoClimaConfigurationBase {
+    public enum ConnectionMode {
+        LOCAL_CONNECTION,
+        REMOTE_API_STUB,
+        REMOTE_API_PROXY
+    }
+
+    public enum DeviceSidePasswordDisplayMode {
+        NEVER,
+        MASKED,
+        CLEARTEXT
+    }
+
+    /**
+     * Argo configuration parameters specific to local connection & API stub
+     * These names are defined in thing-types.xml and get injected on instantiation
+     * through {@link org.openhab.core.thing.binding.BaseThingHandler#getConfigAs getConfigAs}
+     */
```

Comment:
```suggestion
```

default behaviour

Reply:
Ack default. But is it wrong? Occasional developers (like me) reading the code in non-IDE don't necessairly know there's a inject happening. Is documenting something for my own readability benefit a bad practice?

Reply:
Comments can be perfectly fine, especially if they document some specific condition. This is not the case here.

Reply:
Is partially violating Occam's Razor (guilty of that!) enough reason to remove?

---

FYI - reason I added it and included a jump link to the actual function which does it in core is because I'm using array syntax for some params, and these are not exactly perfectly documented by the framework (I ended up debugging XCode files and code generation rules). 
While is not commenting on code I **wrote**, it does leave a breadcrumb about code I **use**. 
This class contains a DTO part which needs to be kept in sync w/ the XML (in ideal world one would generate from the other and/or test sync compile-time, but we don't have this infrastructure here (yet?)).

Long story short, there's at least one person who benefits from this excessive comment (_me - and has everything to do with me not being intimate w/ the framework_) and I think you're also hinting it is not wrong per-se, just superflous.

If you see a way to preserve at least gist of it (in this file) that doesn't raise eyebrows, I'd rather do that. Any wording you think would be OK?
If this is that bad of a practice, let's settle on:
  - I'll keep it in code for all N-1 commits (as it helps me not to forget the exact _what-flows-where_)
  - Will remove in the very last commit of this PR (ex. commit your deletion suggestion directly from GH).

Reply:
In that case please remove before the last commit (about now ;-) )